_posts/2017-08-17-grpc-flatbuffers.md

```diff
@@ -0,0 +1,76 @@
+---
+layout: post
+title: Announcing out of the box support for gRPC in the Flatbuffers serializaton library.
+published: true
+permalink: blog/flatbuffers
+author: Wouter van Oortmerssen
+company: Google
+company-link: https://www.google.com
+---
+
+The recent release of Flatbuffers [version 1.7](https://github.com/google/flatbuffers/releases) introduced truly zero-copy support for gRPC out of the box. 
+
+[Flatbuffers](https://google.github.io/flatbuffers/) is a serialization library that allows you to access serialized data without first unpacking it or allocating any 
+additional data structures. It was originally designed for games and other resource constrained applications, but is now finding use
+within many teams at Google, and outside (eg: Facebook, Netflix).
+
+<!--more-->
+
+Flatbuffers enables maximum throughput by directly using gRPC's slice buffers with zero-copy<sup>1</sup> for the common use cases. An incoming rpc may be processed directly from gRPCs internal buffers, and you constructing new message will write directly to internal buffers without intermediate steps.
+
+So far, this is supported in the C++ implementation of FlatBuffers. There is also an implementation in Go, which is not entirely zero copy, but still very low on allocation cost (see below). Other languages are in the works.
+
+<sup>1</sup>Advanced use cases, such as resizing, will result in copying, zero-copy here refers to the general case.  
```

Comment:
I wouldn't call this Advanced. I'd leave this note out entirely, but if you want to leave it in, I'd write: "Resizing buffers can still cause copying, but this can usually be avoided".